cli: TestPartialZip failed

cli.TestPartialZip [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3367054&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3367054&tab=artifacts#/) on master @ [2b8d42d76aba893dbef2e08b069a5ecf3f6bd566](https://github.com/cockroachdb/cockroach/commits/2b8d42d76aba893dbef2e08b069a5ecf3f6bd566):


```
        [node 3] writing range 41... converting to JSON... writing binary output: debug/nodes/3/ranges/41.json... done
        [node 3] writing range 42... converting to JSON... writing binary output: debug/nodes/3/ranges/42.json... done
        [node 3] writing range 43... converting to JSON... writing binary output: debug/nodes/3/ranges/43.json... done
        [cluster] pprof summary script... writing binary output: debug/pprof-summary.sh... done
        [cluster] hot range summary script... writing binary output: debug/hot-ranges.sh... done
        
    zip_test.go:385: 
        testdata/zip/partial1:1: 
        output didn't match expected:
        @@ -47,12 +47,16 @@
         [node 1] retrieving SQL data for crdb_internal.node_inflight_trace_spans... writing output: debug/nodes/1/crdb_internal.node_inflight_trace_spans.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_metrics... writing output: debug/nodes/1/crdb_internal.node_metrics.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_queries... writing output: debug/nodes/1/crdb_internal.node_queries.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_runtime_info... writing output: debug/nodes/1/crdb_internal.node_runtime_info.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_sessions... writing output: debug/nodes/1/crdb_internal.node_sessions.txt... done
        -[node 1] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/1/crdb_internal.node_statement_statistics.txt... done
        -[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/1/crdb_internal.node_transaction_statistics.txt... done
        +[node 1] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/1/crdb_internal.node_statement_statistics.txt...
        +[node 1] retrieving SQL data for crdb_internal.node_statement_statistics: last request failed: pq: internal error: read-stmt-stats: descriptor not found
        +[node 1] retrieving SQL data for crdb_internal.node_statement_statistics: creating error output: debug/nodes/1/crdb_internal.node_statement_statistics.txt.err.txt... done
        +[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/1/crdb_internal.node_transaction_statistics.txt...
        +[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics: last request failed: pq: internal error: read-txn-stats: descriptor not found
        +[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics: creating error output: debug/nodes/1/crdb_internal.node_transaction_statistics.txt.err.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_transactions... writing output: debug/nodes/1/crdb_internal.node_transactions.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_txn_stats... writing output: debug/nodes/1/crdb_internal.node_txn_stats.txt... done
         [node 1] retrieving SQL data for crdb_internal.active_range_feeds... writing output: debug/nodes/1/crdb_internal.active_range_feeds.txt... done
         [node 1] requesting data for debug/nodes/1/details... received response... converting to JSON... writing binary output: debug/nodes/1/details.json... done
         [node 1] requesting data for debug/nodes/1/gossip... received response... converting to JSON... writing binary output: debug/nodes/1/gossip.json... done
        @@ -213,12 +217,16 @@
         [node 3] retrieving SQL data for crdb_internal.node_inflight_trace_spans... writing output: debug/nodes/3/crdb_internal.node_inflight_trace_spans.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_metrics... writing output: debug/nodes/3/crdb_internal.node_metrics.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_queries... writing output: debug/nodes/3/crdb_internal.node_queries.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_runtime_info... writing output: debug/nodes/3/crdb_internal.node_runtime_info.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_sessions... writing output: debug/nodes/3/crdb_internal.node_sessions.txt... done
        -[node 3] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/3/crdb_internal.node_statement_statistics.txt... done
        -[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/3/crdb_internal.node_transaction_statistics.txt... done
        +[node 3] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/3/crdb_internal.node_statement_statistics.txt...
        +[node 3] retrieving SQL data for crdb_internal.node_statement_statistics: last request failed: pq: internal error: read-stmt-stats: descriptor not found
        +[node 3] retrieving SQL data for crdb_internal.node_statement_statistics: creating error output: debug/nodes/3/crdb_internal.node_statement_statistics.txt.err.txt... done
        +[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/3/crdb_internal.node_transaction_statistics.txt...
        +[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics: last request failed: pq: internal error: read-txn-stats: descriptor not found
        +[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics: creating error output: debug/nodes/3/crdb_internal.node_transaction_statistics.txt.err.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_transactions... writing output: debug/nodes/3/crdb_internal.node_transactions.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_txn_stats... writing output: debug/nodes/3/crdb_internal.node_txn_stats.txt... done
         [node 3] retrieving SQL data for crdb_internal.active_range_feeds... writing output: debug/nodes/3/crdb_internal.active_range_feeds.txt... done
         [node 3] requesting data for debug/nodes/3/details... received response... converting to JSON... writing binary output: debug/nodes/3/details.json... done
         [node 3] requesting data for debug/nodes/3/gossip... received response... converting to JSON... writing binary output: debug/nodes/3/gossip.json... done
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestPartialZip665375686
--- FAIL: TestPartialZip (2.83s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestPartialZip PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server knz
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPartialZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestPartialZip [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3373781&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3373781&tab=artifacts#/) on master @ [45ffbd022fb18124c610a2926dd7028a707f3f45](https://github.com/cockroachdb/cockroach/commits/45ffbd022fb18124c610a2926dd7028a707f3f45):


```
        [node 3] writing range 41... converting to JSON... writing binary output: debug/nodes/3/ranges/41.json... done
        [node 3] writing range 42... converting to JSON... writing binary output: debug/nodes/3/ranges/42.json... done
        [node 3] writing range 43... converting to JSON... writing binary output: debug/nodes/3/ranges/43.json... done
        [cluster] pprof summary script... writing binary output: debug/pprof-summary.sh... done
        [cluster] hot range summary script... writing binary output: debug/hot-ranges.sh... done
        
    zip_test.go:385: 
        testdata/zip/partial1:1: 
        output didn't match expected:
        @@ -47,12 +47,16 @@
         [node 1] retrieving SQL data for crdb_internal.node_inflight_trace_spans... writing output: debug/nodes/1/crdb_internal.node_inflight_trace_spans.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_metrics... writing output: debug/nodes/1/crdb_internal.node_metrics.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_queries... writing output: debug/nodes/1/crdb_internal.node_queries.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_runtime_info... writing output: debug/nodes/1/crdb_internal.node_runtime_info.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_sessions... writing output: debug/nodes/1/crdb_internal.node_sessions.txt... done
        -[node 1] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/1/crdb_internal.node_statement_statistics.txt... done
        -[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/1/crdb_internal.node_transaction_statistics.txt... done
        +[node 1] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/1/crdb_internal.node_statement_statistics.txt...
        +[node 1] retrieving SQL data for crdb_internal.node_statement_statistics: last request failed: pq: internal error: read-stmt-stats: descriptor not found
        +[node 1] retrieving SQL data for crdb_internal.node_statement_statistics: creating error output: debug/nodes/1/crdb_internal.node_statement_statistics.txt.err.txt... done
        +[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/1/crdb_internal.node_transaction_statistics.txt...
        +[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics: last request failed: pq: internal error: read-txn-stats: descriptor not found
        +[node 1] retrieving SQL data for crdb_internal.node_transaction_statistics: creating error output: debug/nodes/1/crdb_internal.node_transaction_statistics.txt.err.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_transactions... writing output: debug/nodes/1/crdb_internal.node_transactions.txt... done
         [node 1] retrieving SQL data for crdb_internal.node_txn_stats... writing output: debug/nodes/1/crdb_internal.node_txn_stats.txt... done
         [node 1] retrieving SQL data for crdb_internal.active_range_feeds... writing output: debug/nodes/1/crdb_internal.active_range_feeds.txt... done
         [node 1] requesting data for debug/nodes/1/details... received response... converting to JSON... writing binary output: debug/nodes/1/details.json... done
         [node 1] requesting data for debug/nodes/1/gossip... received response... converting to JSON... writing binary output: debug/nodes/1/gossip.json... done
        @@ -213,12 +217,16 @@
         [node 3] retrieving SQL data for crdb_internal.node_inflight_trace_spans... writing output: debug/nodes/3/crdb_internal.node_inflight_trace_spans.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_metrics... writing output: debug/nodes/3/crdb_internal.node_metrics.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_queries... writing output: debug/nodes/3/crdb_internal.node_queries.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_runtime_info... writing output: debug/nodes/3/crdb_internal.node_runtime_info.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_sessions... writing output: debug/nodes/3/crdb_internal.node_sessions.txt... done
        -[node 3] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/3/crdb_internal.node_statement_statistics.txt... done
        -[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/3/crdb_internal.node_transaction_statistics.txt... done
        +[node 3] retrieving SQL data for crdb_internal.node_statement_statistics... writing output: debug/nodes/3/crdb_internal.node_statement_statistics.txt...
        +[node 3] retrieving SQL data for crdb_internal.node_statement_statistics: last request failed: pq: internal error: read-stmt-stats: descriptor not found
        +[node 3] retrieving SQL data for crdb_internal.node_statement_statistics: creating error output: debug/nodes/3/crdb_internal.node_statement_statistics.txt.err.txt... done
        +[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics... writing output: debug/nodes/3/crdb_internal.node_transaction_statistics.txt...
        +[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics: last request failed: pq: internal error: read-txn-stats: descriptor not found
        +[node 3] retrieving SQL data for crdb_internal.node_transaction_statistics: creating error output: debug/nodes/3/crdb_internal.node_transaction_statistics.txt.err.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_transactions... writing output: debug/nodes/3/crdb_internal.node_transactions.txt... done
         [node 3] retrieving SQL data for crdb_internal.node_txn_stats... writing output: debug/nodes/3/crdb_internal.node_txn_stats.txt... done
         [node 3] retrieving SQL data for crdb_internal.active_range_feeds... writing output: debug/nodes/3/crdb_internal.active_range_feeds.txt... done
         [node 3] requesting data for debug/nodes/3/details... received response... converting to JSON... writing binary output: debug/nodes/3/details.json... done
         [node 3] requesting data for debug/nodes/3/gossip... received response... converting to JSON... writing binary output: debug/nodes/3/gossip.json... done
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestPartialZip755141975
--- FAIL: TestPartialZip (4.09s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestPartialZip PKG=./pkg/cli TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/server knz
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPartialZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/BUILD.bazel
+++ b/pkg/cli/BUILD.bazel
@@ -337,6 +337,7 @@ go_test(
         "//pkg/sql",
         "//pkg/sql/catalog/descpb",
         "//pkg/sql/protoreflect",
+        "//pkg/sql/tests",
         "//pkg/storage",
         "//pkg/testutils",
         "//pkg/testutils/buildutil",

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -34,6 +34,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/security"
 	"github.com/cockroachdb/cockroach/pkg/server"
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/tests"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
@@ -358,8 +359,10 @@ func TestPartialZip(t *testing.T) {
 	ctx := context.Background()
 
 	// Three nodes. We want to see what `zip` thinks when one of the nodes is down.
+	params, _ := tests.CreateTestServerParams()
+	params.Insecure = true
 	tc := testcluster.StartTestCluster(t, 3,
-		base.TestClusterArgs{ServerArgs: base.TestServerArgs{Insecure: true}})
+		base.TestClusterArgs{ServerArgs: params})
 	defer tc.Stopper().Stop(ctx)
 
 	// Switch off the second node.

